documentation/running_bolt_commands.md

```diff
@@ -1,418 +1,640 @@
+  bolt command run 'pwd' --targets 'bolt*'
+  ```
 
-## Upload files or directories to remote targets
+- _PowerShell cmdlet_
 
-Use Bolt to copy files or directories to remote targets.
+  ```powershell
+  Invoke-BoltCommand -Command 'pwd' -Targets 'bolt*'
+  ```
 
-**Note:** Most transports are not optimized for file copying, so this command is
-best limited to small files.
+#### Reading targets from a file
 
--   To upload a file or directory to a remote target, run the `bolt file upload`
-    command. Specify the local path to the file or directory, the destination
-    location, and the targets.
+You can also read a list of targets from a file. To read a file of targets,
+pass an `@` symbol, followed by the relative path to the file, to the
+`targets` option.
 
-    ```
-    bolt file upload <SOURCE> <DESTINATION> --targets <TARGET NAME>,<TARGET NAME>
-    ```
+- _\*nix shell command_
 
-    ```
-    bolt file upload my_file.txt /tmp/remote_file.txt --targets web5.mydomain.edu,web6.mydomain.edu
-    ```
+  ```shell
+  bolt command run 'pwd' --targets '@targets.txt'
+  ```
 
-## Download files or directories from remote targets
+- _PowerShell cmdlet_
 
-Use Bolt to copy files or directories from remote targets to your local system.
+  ```powershell
+  Invoke-BoltCommand -Command 'pwd' -Targets '@targets.txt'
+  ```
 
-To download a file or directory from a remote target, run the
-`bolt file download` command. Specify the remote path to the file or
-directory, the destination directory on your local system, and the targets.
+  > **Note:** In PowerShell, always wrap the file name in single quotes.
 
-```shell
-$ bolt file download <SOURCE> <DESTINATION> --targets <TARGETS>
-```
+#### Reading targets from `stdin`
 
-The `destination` can be either an absolute or relative path to a directory
-on your local system. If you use a relative path, Bolt expands the path
-relative to the current working directory. If the destination directory does
-not exist, Bolt will automatically create it.
+Targets can also be read from `stdin`. This can be helpful for piping
+results from another command to Bolt. To read a list of targets from `stdin`,
+pipe the results from another command to Bolt and pass a single dash `-` to
+the `targets` option.
 
-Each file or directory is saved to the destination directory under a
-directory with a name matching the URL-encoded name of the target it
-was downloaded from. The target directory names are URL-encoded to ensure
-that they are valid directory names.
+- _\*nix shell command_
 
-For example, the following command downloads the SSH daemon configuration 
-file from two targets, `linux` and `ssh://example.com`, and saves it to the
-destination directory `sshd_config`:
+  ```shell
+  cat targets.txt | bolt command run 'pwd' --targets -
+  ```
 
-```shell
-$ bolt file download /etc/ssh/sshd_config sshd_config --targets linux,ssh://example.com
-```
+> **Note:** Reading from `stdin` is not supported by the PowerShell module.
```

Comment:
```suggestion
Reading from `stdin` is not supported by the PowerShell module.
```
I'm trimming down the notes on the page a bit. If we have too many it gets a bit noisy.